cvat/apps/engine/views.py

```diff
@@ -3376,104 +3394,117 @@ def _import_annotations(request, rq_id_factory, rq_func, db_obj, format_name,
         rq_id = rq_id_factory(db_obj.pk).render()
 
     queue = django_rq.get_queue(settings.CVAT_QUEUES.IMPORT_DATA.value)
-    rq_job = queue.fetch_job(rq_id)
 
-    if rq_job and rq_id_should_be_checked and not is_rq_job_owner(rq_job, request.user.id):
-        return Response(status=status.HTTP_403_FORBIDDEN)
+    # ensure that there is no race condition when processing parallel requests
+    with get_rq_lock_for_job(queue, rq_id):
```

Comment:
Since I've refctored a bit the code to reduce the critical section, I decided to mark `PUT /api/tasks|jobs/id/annotations?rq_id=rq_id` as deprecated in this PR